x/poe/types/genesis.go

```diff
@@ -16,240 +16,242 @@ const DefaultBondDenom = "utgd"
-				Threshold:     sdk.NewDec(66),
-				AllowEndEarly: true,
+			EngagementContractConfig: &EngagementContractConfig{
+				Halflife: 180 * 24 * time.Hour,
 			},
-		},
-		ArbiterPoolContractConfig: &ArbiterPoolContractConfig{
-			Name:         "Arbiter Pool",
-			EscrowAmount: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
-			VotingRules: VotingRules{
-				VotingPeriod:  30,
-				Quorum:        sdk.NewDec(51),
-				Threshold:     sdk.NewDec(55),
-				AllowEndEarly: true,
+			OversightCommitteeContractConfig: &OversightCommitteeContractConfig{
+				Name:         "Oversight Community",
+				EscrowAmount: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+				VotingRules: VotingRules{
+					VotingPeriod:  30,
+					Quorum:        sdk.NewDec(51),
+					Threshold:     sdk.NewDec(55),
+					AllowEndEarly: true,
+				},
+			},
+			CommunityPoolContractConfig: &CommunityPoolContractConfig{
+				VotingRules: VotingRules{
+					VotingPeriod:  21,
+					Quorum:        sdk.NewDec(10),
+					Threshold:     sdk.NewDec(60),
+					AllowEndEarly: true,
+				},
+			},
+			ValidatorVotingContractConfig: &ValidatorVotingContractConfig{
+				VotingRules: VotingRules{
+					VotingPeriod:  14,
+					Quorum:        sdk.NewDec(40),
+					Threshold:     sdk.NewDec(66),
+					AllowEndEarly: true,
+				},
 			},
-			DisputeCost: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+			ArbiterPoolContractConfig: &ArbiterPoolContractConfig{
+				Name:         "Arbiter Pool",
+				EscrowAmount: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+				VotingRules: VotingRules{
+					VotingPeriod:  30,
+					Quorum:        sdk.NewDec(51),
+					Threshold:     sdk.NewDec(55),
+					AllowEndEarly: true,
+				},
+				DisputeCost: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+			},
+			SystemAdminAddress: sdk.AccAddress(rand.Bytes(address.Len)).String(),
 		},
-		SystemAdminAddress: sdk.AccAddress(rand.Bytes(address.Len)).String(),
-		Params:             DefaultParams(),
 	}
 }
 
+// ValidateGenesis validates genesis for PoE module
 func ValidateGenesis(g GenesisState, txJSONDecoder sdk.TxDecoder) error {
-	if g.SeedContracts {
+	if err := g.Params.Validate(); err != nil {
+		return sdkerrors.Wrap(err, "params")
+	}
+	if g.SeedContracts != nil {
+		// seed genesis
 		if len(g.Contracts) != 0 {
 			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "seed enabled but PoE contracts addresses provided")
 		}
-		if len(g.Engagement) == 0 {
-			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty engagement group")
+		if err := validateSeedContracts(g.SeedContracts, txJSONDecoder); err != nil {
+			return sdkerrors.Wrap(err, "seed contracts")
 		}
-		if g.EngagementContractConfig == nil {
-			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty engagement contract config")
-		}
-		if err := g.EngagementContractConfig.ValidateBasic(); err != nil {
-			return sdkerrors.Wrap(err, "engagement contract config")
+		return nil
```

Comment:
Should we error here if both are set? Or neither is set?

`(g.SeedContracts != nil) == (len(g.Contracts) > 0)`

That would be nice in the validation step at least to avoid some confusion.

Reply:
yes, this check is some lines above